src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +160,60 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    /// <summary>
+    /// Checks the merge element for a frequency attribute. Then determines whether the current run
+    /// matches the frequency criteria. Valid frequency values are 'daily' and 'weekly'.
+    /// </summary>
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun, DateTime runDateTime)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // Throw when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            throw new Exception($"Unexpected merge frequency specified: '{frequency}'. Valid values are 'daily' and 'weekly'.");
+        }
+
+        // Since cron should schedule this to run every 3 hours starting at 12am,
+        // we expect to be run within a 10 minute window of this time.
+
+        // Adjust the time 5 minutes into the future in case the pipeline machine
+        // and scheduler machine have mismatched clocks.
+        var adjustedRunDateTime = runDateTime.AddMinutes(5);
+        var adjustedRunDate = adjustedRunDateTime.Date;
+
+        // Because the adjusted run time is being used, we treat a run as valid if
```

Comment:
was this heuristic decided on by looking at the actual start times for the pipeline?

Reply:
Paranoia. The pipeline seems to be very consistent from a cursory look at start times.